internal/exec/stages/disks/partitions.go

```diff
@@ -353,7 +354,14 @@ func (s stage) partitionDisk(dev types.Disk, devAlias string) error {
 		case exists && shouldExist && matches:
 			s.Logger.Info("partition %d found with correct specifications", part.Number)
 		case exists && shouldExist && !wipeEntry && !matches:
-			return fmt.Errorf("Partition %d didn't match: %v", part.Number, matchErr)
+			if resize {
```

Comment:
In this case we need to check that everything else in the struct matches, other than the partition size.